lesson17/src/js/main.js

```diff
@@ -0,0 +1,212 @@
+function showLoadingImage() {
+  const main = document.getElementById("main");
+  const fragment = document.createDocumentFragment();
+  const renderLoadingBox = document.createElement("div");
+  renderLoadingBox.classList.add("loading");
+  renderLoadingBox.id = "js-loading";
+  const loadingImage = document.createElement("img");
+  loadingImage.src = "images/loading.gif";
+  loadingImage.width = 80;
+  loadingImage.height = 80;
+  loadingImage.classList.add("loading__image");
+  fragment.appendChild(renderLoadingBox).appendChild(loadingImage);
+  document.body.insertBefore(fragment, main);
+}
+
+function removeLoading() {
+  document.getElementById("js-loading").remove();
+}
+
+function displayErrorMassage(error) {
+  const slideImageArea = document.querySelector(".mainvisual__images__inner");
+  const textBox = document.createElement("p");
+  textBox.classList.add("");
+  textBox.textContent = error;
+  slideImageArea.appendChild(textBox);
+}
+
+function renderSlideArea() {
+  const slideImageArea = document.createElement("div");
+  slideImageArea.classList.add("mainvisual__images__inner");
+
+  document.querySelector(".mainvisual__images").appendChild(slideImageArea);
+}
+
+function renderSlideImage(item) {
+  const slideImageArea = document.querySelector(".mainvisual__images__inner");
+
+  const slider = document.createElement("ul");
+  slider.classList.add("slider");
+
+  const fragment = document.createDocumentFragment();
+  item.forEach((image, i) => {
+    const sliderItem = document.createElement("li");
+    sliderItem.classList.add("slider__item");
+    sliderItem.setAttribute("data-slide-index", i + 1);
+
+    const sliderImage = document.createElement("img");
+    sliderImage.classList.add("slider__image");
+    sliderImage.src = image.image;
+    sliderImage.width = image.width;
+    sliderImage.height = image.height;
+
+    if (i === 0) sliderItem.classList.add("is-active");
+
+    fragment.appendChild(sliderItem).appendChild(sliderImage);
+  });
+
+  slideImageArea.appendChild(slider).appendChild(fragment);
+}
+
+function renderSlideButton() {
+  let fragment = document.createDocumentFragment();
+
+  const buttonArea = document.createElement("div");
+  buttonArea.classList.add("slider__button");
+
+  const prevButton = document.createElement("button");
+  prevButton.classList.add("arrow", "prev");
+  prevButton.id = "js-button_prev";
+  prevButton.setAttribute("aria-label", "previous");
+  prevButton.textContent = "previous";
+
+  fragment.appendChild(prevButton);
+
+  const nextButton = document.createElement("button");
+  nextButton.classList.add("arrow", "next");
+  nextButton.id = "js-button_next";
+  nextButton.textContent = "next";
+  nextButton.setAttribute("aria-label", "next");
+
+  fragment.appendChild(nextButton);
+
+  const slideArea = document.querySelector(".mainvisual__images");
+  slideArea.appendChild(buttonArea).appendChild(fragment);
+}
```

Comment:
```suggestion
  const buttonArea = document.createElement("div");
  buttonArea.classList.add("slider__button");
  const direction = ["prev", "next"];
  direction.forEach((element) => {
    const button = document.createElement("button");
    button.classList.add("arrow", `${element}`);
    button.id = `js-button_${element}`;
    button.textContent = `${element}`;
    button.setAttribute("aria-label", `${element}`);
    fragment.appendChild(button);
  });
  const slideArea = document.querySelector(".mainvisual__images");
  slideArea.appendChild(buttonArea).appendChild(fragment);
}
```
IMO
Looping through the use of forEach allows duplicated code to be combined.

Reply:
With forEach, the code is pretty organized!
I'm not good at writing variables by assigning them to characters, so I'll use this as an opportunity!

[4be59a5](https://github.com/nyattsu72/hanson-frontend-cource/pull/25/commits/4be59a53dbad8e67a56322f80d680317bfb9eb0c)